src/Sylius/Bundle/ProductBundle/Doctrine/ORM/ProductRepository.php

```diff
@@ -43,6 +43,7 @@ public function findByNamePart(string $phrase, string $locale): array
             ->andWhere('translation.name LIKE :name')
             ->setParameter('name', '%' . $phrase . '%')
             ->setParameter('locale', $locale)
+            ->setMaxResults(25)
```

Comment:
Maybe make it an optional argument with 25 as default value?

Reply:
Nevermind, this is an interface method.

Reply:
My idea would be to add an optional parameter to the method definition:

```
public function findByNamePart(string $phrase, string $locale, ?int $limit = null): array
```

This way, the behaviour wouldn't change, but we'd allow for specifying the max results amount.

The second step to actually make use of it would be to modify the routing of the action which provides autocompletion (`sylius_admin_ajax_product_by_name_phrase`) and limit the query to 25 there.

It would be best if we could change it for both `ProductRepositoryInterface` and `TaxonRepositoryInterface`, they both have the same method with the same usage.

What do you think?

Reply:
@pamil That's what I thought of too, but wouldn't it be a BC break?

Reply:
Yeap, as I said - this could be improved by adding the default limit to the signature. However, this would be a BC if someone customized the AJAX route in his local environment. As this would be a bugfix, I think that it makes more sense to add the limit in the default repository interface like this: 
`public function findByNamePart(string $phrase, string $locale, ?int $limit = 25): array`

Reply:
Even if the route got overwritten, it should then default to not including the limit - tested this on Sylius-Standard.